src/main/groovy/com/google/gcloud/task/GCloudTask.groovy

```diff
@@ -1,45 +1,118 @@
-        GCloudPluginExtension gcEx = project.extensions.getByName("gcloud")
-        if (gcEx.gcloudHome) {
-            gcloud.setGCloudHome(new File(gcEx.gcloudHome));
+    protected void start() {
+        validateCloudSdk()
+        executeTask()
+    }
+
+    void validateCloudSdk() {
+        // TODO: validate the cloud sdk install somehow here
+    }
+
+    // automatically find the gcloud executable
+    private String getGCloudCommand() {
+        def gcloudHomeSysProp = System.getProperty(GCLOUD_HOME_SYS_PROP_KEY)
+        if (gcloudHomeSysProp) {
+            File gcloudBin = new File(gcloudHomeSysProp, "bin/gcloud")
+            if (!gcloudBin.exists()) {
+                logger.warn("WARNING: gcloud not found in " + gcloudBin + ", have you set ${GCLOUD_HOME_SYS_PROP_KEY} correctly?");
+                logger.warn("looking for gcloud in other locations")
+            }
+            else {
+                return gcloudBin.absolutePath
+            }
+        }
+
+        // get gcloud path from env var
+        def gcloudHomeEnvVar = System.getenv(GCLOUD_HOME_ENV_VAR_KEY)
+        if (gcloudHomeEnvVar) {
+            File gcloudBin = new File(gcloudHomeEnvVar, "bin/gcloud")
+            if (!gcloudBin.exists()) {
+                logger.warn("WARNING: gcloud not found in " + gcloudBin + ", have you set ${GCLOUD_HOME_ENV_VAR_KEY} correctly?");
+                logger.warn("looking for gcloud in other locations")
+            } else {
+                return gcloudBin.absolutePath
+            }
         }
-        println(getCommand());
-        if (isRunAsync) {
-            gcloud.runAsync(null)
+
+        return "gcloud";
+    }
+
+    // override this to add more parts to the command
+    protected void configureCommand() {
+        // add the gcloud command
+        commandBuilder.addCommand(getGCloudCommand())
+
+        // add subcommand parts
+        getCommand()?.each { part ->
+            commandBuilder.addCommand(part)
         }
-        else {
-            gcloud.runSync()
+
+        // add all options/args
+        getArgs()?.each { key, val ->
+            if (val instanceof Collection) {
+                commandBuilder.addOption(key, val)
+            }
+            else {
+                commandBuilder.addOption(key, val.toString())
+            }
         }
     }
 
-    protected String[] getCommand() {
-        return command;
+    protected void executeTask() {
+        ProcessRunner pr = new GradleExecProcessRunner(getCommandLine(), project)
+        int exitCode = pr.run()
+        if (exitCode != 0) {
+            throw new GradleException("gcloud failed with code " + exitCode)
+        }
     }
 
-    protected void addBaseCommandOptions(GCloudCommandBuilder builder) {
-        GCloudPluginExtension gcEx = project.extensions.getByName("gcloud")
-        builder.addOption("project", gcEx.project)
-               .addOption("verbosity", gcEx.verbosity)
+
+    // NOT idempotent, ensure single call
```

Comment:
actually never mind, there was no need to make CommandBuilder a member of the class, just creating a new instance and passing it around now.